Feliz.CompilerPlugins/ReactComponent.fs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Feliz
+
+open Fable
+open Fable.AST
+
+/// <summary>Transforms a function into a React function component. Make sure the function is defined at the module level</summary>
+type ReactComponentAttribute() =
+    inherit MemberDeclarationPluginAttribute()
+    override _.FableMinimumVersion = "3.0"
+
+    /// <summary>Transforms call-site into createElement calls</summary>
+    override _.TransformCall(logger, memb, expr) =
+        let membArgs = memb.CurriedParameterGroups |> List.concat
+        match expr with
+        | Fable.Call(callee, info, typeInfo, range) when List.length membArgs = List.length info.Args ->
+            let callee =
+                match callee with
+                | Fable.Expr.IdentExpr ident -> Fable.Expr.IdentExpr { ident with Name = AstUtils.capitalize ident.Name }
+                | _ -> callee
+
+            if info.Args.Length = 1 && AstUtils.isRecord info.Args.[0].Type then
+                // F# Component { Value = 1 }
+                // JSX <Component Value={1} />
+                // JS createElement(Component, { Value: 1 })
+                AstUtils.makeCall (AstUtils.makeImport "createElement" "react") [callee; info.Args.[0]]
+            elif info.Args.Length = 1 && info.Args.[0].Type = Fable.Type.Unit then
+                // F# Component()
+                // JSX <Component />
+                // JS createElement(Component, null)
+                AstUtils.makeCall (AstUtils.makeImport "createElement" "react") [callee; AstUtils.nullValue]
+            else
+            let propsObj =
+                List.zip membArgs info.Args
+                |> List.choose (fun (arg, expr) -> arg.Name |> Option.map (fun k -> k, expr))
+                |> AstUtils.objExpr
+
+            // TODO: Detect usage of an input called `children` anc spread it into the component
```

Comment:
At the moment this is complicated because the argument info is not included in the `Fable.Call` expression. In the next update probably we should include the MemberInfo (or parts of it) in CallInfo.

Reply:
That would be great to have!